File: 724-find-pivot-index/find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        r=len(nums)-1
        n=[0]*len(nums)
        total=0
        l=0
        for i in range(len(nums)):
            total=total+nums[i]
            n[i]=total
        while l <= r:
            if l == 0 and (n[r] - n[l]) == 0:
                return 0
            elif l != 0 and n[l-1] == (n[r] - n[l]):
                return l
            l += 1
        
        return -1
            

        

File: 724-find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n=nums.length;
        int[] a=new int[n];
        int sum=0;
      for(int i=0;i<n;i++){
         sum+=nums[i];
         a[i]=sum;
      }  
      for(int j=0;j<n;j++){
        if(j==0 && (a[n-1]-a[0])==0) return 0;
        else if(j==n-1 && a[j-1]==0) return n-1;
        else if(j!=0 && a[j]==(a[n-1]-a[j-1])) return j;  
        }
    return -1;
    }
}